### ISSUE TITLE
ViewComponentTagHelper: Allow omitting attribute for parameters with default value

### ISSUE THREAD

**Issue by diryboy** [aspnet/Razor#1266]
Say we have a ViewComponent class

```C#
class MyViewComponent
{
    IViewComponentResult Invoke( bool showSomething = false ) { ... }
}
```

It would be great to just only write `<vc:my />` in Razor if one don't want to show that something. Currently you have to write `<vc:my show-something="false" />`.

**Comment by DamianEdwards** [comment]
Agreed. No idea how feasible it is, but I agree with the sentiment :smile:

**Comment by rynowak** [comment]
This seems pretty feasible. We need to teach [this code](https://github.com/aspnet/Razor/blob/dev/src/Microsoft.CodeAnalysis.Razor/ViewComponentTagHelperDescriptorFactory.cs#L62) about default values. Currently it says that each parameter is a *required* attribute

**Comment by DamianEdwards** [comment]
@rynowak candidate for preview2?

**Comment by rynowak** [comment]
I'm going to move it in and mark it up for grabs. At this point locking down our public APIs and completing the tooling story needs to take priority.

**Comment by svallis** [comment]
Really looking forward to this fix. As things stand, using the VC tag helper is very dangerous. Add a new parameter to an existing and widely used tag helper and miss updating one of the places it is called and it _silently fails while emitting server side markup to the browser_. I guess we're too far along now for this to make it in to 2.0.0?

**Comment by rynowak** [comment]
This is not going to make it for 2.0.0. We don't have time to update tooling at this point.

**Comment by Liero** [comment]
I believe it should be marked as a bug:

This is working:

    @await Component.Invoke(typeof(MyViewComponent));

This is not working 

    <vc:My />

It is expected behavior and since you do not show any warning, error, debug message, nothing, it upsets a lot of people wondering why the hell the view component is not rendered

**Comment by clockwiseq** [comment]
So will this make it into the next release?
